docs/api/embeddings.md

```diff
@@ -1,26 +1,132 @@
 ---
 id: embeddings
-title: Embeddings
+title: Embeddings & RAG
 ---
 
+Large Language Models (LLMs) are powerful, but they **do not know your private or internal data**.
+They are trained on public information and cannot access your documents, databases, or source
+code unless you explicitly provide that context.
+
+This is the problem that **Embeddings** and **Retrieval-Augmented Generation (RAG)** address.
+
+> **Cube AI scope**
+>
+> Cube AI provides **secure embeddings generation and LLM execution inside Trusted Execution Environments (TEEs)**.
+> It does **not** provide a built-in vector database or retrieval layer.
+> In a RAG setup, storing embeddings and retrieving relevant context is handled
+> by the application or integration layer outside of Cube AI.
+
+---
+
+## The Problem LLMs Have
+
+Without embeddings and RAG:
+
+- LLMs cannot answer questions about private data
+- responses are often generic or inaccurate
+- models may hallucinate answers
+- updating knowledge requires retraining (slow and expensive)
+
+---
+
+## What Are Embeddings?
+
+An **embedding** is a numerical (vector) representation of text that captures its meaning.
+
+Embeddings allow applications using Cube AI to:
+
+- compare text by semantic similarity
+- search documents by meaning instead of keywords
+- retrieve relevant context for LLM prompts
+
+In simple terms:
+
+> Embeddings make text **searchable by meaning**, not just by keywords.
+
+Cube AI embeddings are generated inside **Trusted Execution Environments (TEEs)**,
+ensuring that both input text and resulting vectors remain confidential.
```

Comment:
Add a bit more context to it. Link to TEE, a brief explanation how it protects workload and why it's important.

Reply:
Move this to the later section, so we have a flow:
- RAG intro
- RAG explanation
- TEEs & how they help RAG and prompts